common/pkgfiles/referencedfilelist.cpp

```diff
@@ -1104,31 +1144,36 @@ void ReferencedFileList::resolveFiles(const StringArray &locations, const char *
     srcCluster.set(_srcCluster);
     remotePrefix.set(_remotePrefix);
 
-    if (useRemoteStorage)
-    {
-        DBGLOG("ReferencedFileList resolving remote storage files at %s", nullText(remoteLocation));
-        if (!user)
-            user.setown(createUserDescriptor());
-        remoteStorage.set(remoteLocation);
-        ReferencedFileIterator files(this);
-        ForEach(files)
-            files.queryObject().resolveLocalOrRemote(locations, srcCluster, user, remoteStorage, remotePrefix, checkLocalFirst, expandSuperFiles ? &subfiles : NULL, trackSubFiles, resolveLFNForeign);
-    }
-    else
+    ReferencedFileIterator files(this);
+    ForEach(files)
     {
-        if (!isEmptyString(remoteLocation))
-            DBGLOG("ReferencedFileList resolving remote dali files at %s", remoteLocation);
+        ReferencedFile &file = files.queryObject();
+        if (!file.remoteStorage.isEmpty() || useRemoteStorage)
+        {
+            DBGLOG("ReferencedFileList resolving remote storage files at %s", nullText(remoteLocation));
+            if (!user)
+                user.setown(createUserDescriptor());
+            if (!file.remoteStorage.isEmpty()) // Can be set at multiple levels in a packagemap
+                remoteStorage.set(file.remoteStorage);
+            else
+                remoteStorage.set(remoteLocation);
+            file.resolveLocalOrRemote(locations, srcCluster, user, remoteStorage, remotePrefix, checkLocalFirst, expandSuperFiles ? &subfiles : NULL, trackSubFiles, resolveLFNForeign);
+            remoteStorage.clear();
```

Comment:
Does the ability to clear this here mean that remoteStorage is a class member variable that is just being used here for temporary storage?  Is that necessary?

Reply:
I will re-confirm that this is the case, but I recall that `resolveLocalOrRemote` uses the class member variable for its work. But then I was concerned about side-effects since it wasn't previously set at this location. Perhaps it isn't necessary and should be a local temp.